### PR TITLE
feat(changelog): add merge-prerelease flag

### LINKED ISSUE #685: `codecov` dev dependency is deprecated
### Description

`codecov` has been deprecated a while ago and should be removed or replaced.

### Possible Solution

Follow [migration guide](https://docs.codecov.com/docs/deprecated-uploader-migration-guide#python-uploader)

### Additional context

https://github.com/codecov/codecov-python

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MoritzBoehme**
## Description

Added a `--merge-prerelease` flag and config option as described in #687.

## Checklist

- [x] Add test cases to all the changes you introduce
- [x] Run `./scripts/format` and `./scripts/test` locally to ensure this change passes linter check and test
- [x] Test the changes on the local machine manually
- [x] Update the documentation for the changes

## Additional context

The dependency `codecov` seems to be no longer available. ~~I can try to fix it in this pull request, but could also create a new one, depending on what is preferred.~~
I have now to my own frustration fixed the `codecov` action in the second commit. This also closes #685

### CODE REVIEW

File: commitizen/cli.py [hunk 1]
```diff
@@ -247,10 +247,19 @@
                         "name": "--start-rev",
                         "default": None,
                         "help": (
-                            "start rev of the changelog."
+                            "start rev of the changelog. "
```
Comment (MoritzBoehme) [inline commitizen/cli.py:250]: Hope it's fine to sneak this small fix in here. Just something I noticed.

**Comment by MoritzBoehme** [comment]
It seems the pre-commit hook `prettier` is failing for windows only...
No idea, why this would be the case :/

**Comment by MoritzBoehme** [comment]
@woile could you perhaps have a look at the pr? Do you maybe have some idea on how to keep `prettier` from failing?
Thanks for this awesome project btw =)

File: tests/test_changelog.py [hunk 2]
```diff
@@ -800,6 +1080,16 @@ def test_generate_tree_from_commits(gitcommits, tags):
...
 
 
+def test_generate_tree_from_commits_merge_prerelease(gitcommits, tags):
+    parser = defaults.commit_parser
+    changelog_pattern = defaults.bump_pattern
+    tree = changelog.generate_tree_from_commits(
+        gitcommits, tags, parser, changelog_pattern, merge_prerelease=True
+    )
+
+    assert tuple(tree) == COMMITS_TREE_MERGED_PRERELEASES
```
Comment (woile) [inline tests/test_changelog.py:1090]: Should `COMMITS_TREE_MERGED_PRERELEASES` be renamed to `COMMITS_TREE_AFTER_MERGED_PRERELEASES`? 

How is this test different than just `test_generate_tree_from_commits`?

**Comment by woile** [comment]
How does this feature affect existing changelogs?
Will it update old pre-releases as well? could we add a test for it?

**Comment by woile** [comment]
Can you point this PR to `v3`? thanks!

File: tests/test_changelog.py (hunk 2 above)
Comment (MoritzBoehme) [reply tests/test_changelog.py:1090]: Yeah, that name is better. It's different in the sense, that `COMMITS_TREE_AFTER_MERGED_PRERELEASES` does not contain pre-releases (there were 2) and the changes of them are now part of the next non-prerelease.

**Comment by MoritzBoehme** [comment]
> How does this feature affect existing changelogs? Will it update old pre-releases as well? could we add a test for it?

Existing changelogs will be changed when run with this flag. This is unless you also add the `--incremental`, set the `changelog_incremental` config option to `true` or use `cz bump --changelog` (as this implies incremental). I am adding a test right now to ensure this leaves prior pre-releases untouched this way.

> Can you point this PR to v3? thanks!

Will do!

**Pull request merged**